src/statistics.rs

```diff
@@ -0,0 +1,771 @@
+    context: &Context,
+    last_counted_msg: u32,
+    one_one_chats: bool,
+) -> Result<MessageStats> {
+    ensure!(
+        last_counted_msg >= 9,
+        "Last_msgid < 9 would mean including 'special' messages in the statistics"
+    );
+
+    let statistics_bot_chat_id = get_statistics_chat_id(context).await?;
+
+    let trans_fn = |t: &mut rusqlite::Transaction| {
+        t.pragma_update(None, "query_only", "0")?;
+
+        // This table will hold all empty chats,
+        // i.e. all chats that do not contain any members except for self.
+        // Messages in these chats are not actually sent out.
+        t.execute(
+            "CREATE TEMP TABLE temp.empty_chats (
+                id INTEGER PRIMARY KEY
+            ) STRICT",
+            (),
+        )?;
+
+        // id>9 because chat ids 0..9 are "special" chats like the trash chat,
+        // and contact ids 0..9 are "special" contact ids like the 'device'.
+        t.execute(
+            "INSERT INTO temp.empty_chats
+            SELECT id FROM chats
+            WHERE id>9 AND NOT EXISTS(
+                SELECT *
+                FROM contacts, chats_contacts
+                WHERE chats_contacts.contact_id=contacts.id AND chats_contacts.chat_id=chats.id
+				AND contacts.id>9
+            )",
+            (),
+        )?;
+
+        // This table will hold all verified chats,
+        // i.e. all chats that only contain verified contacts.
+        t.execute(
+            "CREATE TEMP TABLE temp.verified_chats (
+                id INTEGER PRIMARY KEY
+            ) STRICT",
+            (),
+        )?;
+
+        // Verified chats are chats that are not empty,
+        // and do not contain any unverified contacts
+        t.execute(
+            "INSERT INTO temp.verified_chats
+            SELECT id FROM chats
+            WHERE id>9
+            AND id NOT IN (SELECT id FROM temp.empty_chats)
+            AND NOT EXISTS(
+                SELECT *
+                FROM contacts, chats_contacts
+                WHERE chats_contacts.contact_id=contacts.id AND chats_contacts.chat_id=chats.id
+				AND contacts.id>9
+				AND contacts.verifier=0
+            )",
+            (),
+        )?;
+
+        // This table will hold all 1:1 chats.
+        t.execute(
+            "CREATE TEMP TABLE temp.one_one_chats (
+                id INTEGER PRIMARY KEY
+            ) STRICT",
+            (),
+        )?;
+
+        t.execute(
+            "INSERT INTO temp.one_one_chats
+            SELECT id FROM chats
+            WHERE type=?;",
+            (Chattype::Single,),
+        )?;
+
+        // - `from_id=?` is to count only outgoing messages.
+        // - `chat_id<>?` excludes the chat with the statistics bot itself,
+        // - `id>?` excludes messages that were already counted in the previously sent statistics, or messages sent before the config was enabled
+        // - `hidden=0` excludes hidden system messages, which are not actually shown to the user
```

Comment:
As for received messages, only reactions are hidden. Probably it's fine to not count reactions and just explain this in the comment. Overall they're also messages

Reply:
> As for received messages, only reactions are hidden.

FTR, we're not counting received messages, we're only counting outgoing messages.

But yes, I'll put a comment.